tests/Unit/TestHelpers.hpp

```diff
@@ -344,17 +344,27 @@ void test_throw_exception(const ThrowingFunctor& func,
   }
 }
 
+/// \cond
+#define MAKE_GENERATOR_IMPL_NAME(NAME, ...) NAME
+#define MAKE_GENERATOR_IMPL_SEED(NAME, SEED, ...) (SEED)
+/// \endcond
+
 /// \ingroup TestingFrameworkGroup
-/// \brief Make a generator of type `std::mt19937`, stores it in a declared
-/// variable of name `NAME`
+/// \brief `MAKE_GENERATOR(NAME [, SEED])` declares a variable of name `NAME`
+/// containing a generator of type `std::mt19937`.
 ///
 /// \details As the generator is made, `INFO` is called to make sure failed
-/// tests provide seed information.
-#define MAKE_GENERATOR(NAME) \
-  std::random_device r;      \
-  const auto seed = r();     \
-  INFO("Seed is: " << seed); \
-  auto(NAME) = std::mt19937 { seed }
+/// tests provide seed information.  `SEED` is chosen randomly if not supplied,
+/// otherwise it must be a constant expression.
+#define MAKE_GENERATOR(...)                                                 \
+  std::mt19937 MAKE_GENERATOR_IMPL_NAME(__VA_ARGS__, DUMMY_TOKEN);          \
+  INFO("Seed is: " << [&MAKE_GENERATOR_IMPL_NAME(__VA_ARGS__,               \
+                                                 DUMMY_TOKEN)]() noexcept { \
+    const auto seed = MAKE_GENERATOR_IMPL_SEED(                             \
+        __VA_ARGS__, std::random_device{}(), DUMMY_TOKEN);                  \
```

Comment:
- "This is cute" -- @fmahebert
- "This is adorable" -- @moxcodes
- "I don't like this" -- @markscheel 

Perhaps this could be slightly easier to understand with a comment explaining what is happening on this line? Or perhaps renaming the macros to `MAKE_GENERATOR_IMPL_FIRST_ARG` and `MAKE_GENERATOR_IMPL_SECOND_ARG`?

(This is a suggestion not a change request, so act as you prefer).